draft-ietf-quic-qlog-main-schema.md

```diff
@@ -1137,18 +1143,72 @@ SimulationMarker = {
 ~~~
 {: #simulation-marker-def title="SimulationMarker definition"}
 
-# Event definition guidelines
+# Event extensibility {#event-extensibility}
+
+The main qlog schema defined by this document describes log and trace formats
+that have related events. A minimal set of common events are presented in
+{{events}}. It is expected that logging is extended with specific, per-protocol
+event definitions that specify the name (category + type) and data needed for
+each individual event. Examples include the QUIC event definitions {{QLOG-QUIC}}
+and HTTP/3 event definitions {{QLOG-H3}}.
+
+New event definitions SHOULD follow the guidance in this section, which provides
+consistency that makes it easier for qlog implementers to support a wide variety
+of protocols.
+
+## Extended Event Schema
+
+New event definitions SHOULD be part of an extension schema, using the
+annotations and concepts presented in this section.
+
+A schema defines one or more qlog event categories that can contain multiple
+event definitions. Each event MUST only belong to a single category.
 
-This document defines the main schema for the qlog format together with some
-common events, which on their own do not provide much logging utility. It is
-expected that logging is extended with specific, per-protocol event definitions
-that specify the name (category + type) and data needed for each individual
-event. Examples include the QUIC event definitions {{QLOG-QUIC}} and HTTP/3
-event definitions {{QLOG-H3}}.
+It is not possible to extend a category with new events. Instead, extension
+schema MUST define new categories with globally unique identifiers. Per
+{{{#name-field}}}, event names are the concatenation of category and type.
+Therefore, unique category identifiers ensure events in qlog files are
+disambuiguated.
+
+Each extension schema is named by a URI. That URI MUST be absolute; it precisely
+identifies the format and meaning of the extension. URIs that contain a domain
```

Comment:
Not exactly sure what the intention is of including "format" here... might be confused with "serialization format", which is certainly not the intent. Replace with a different word or remove?

Reply:
lol, still a hangover from copy/paste. What we should say here is that a schema URI precisely identifies the category name and the event type(s) (the CDDL definition)